zebra-network/src/peer_set/set.rs

```diff
@@ -486,32 +544,44 @@ where
         req: Request,
         hash: InventoryHash,
     ) -> <Self as tower::Service<Request>>::Future {
-        let peer = self
+        let inventory_peer_list = self
             .inventory_registry
             .peers(&hash)
-            .find(|&key| self.ready_services.contains_key(key))
-            .cloned();
+            .filter(|&key| self.ready_services.contains_key(key))
+            .cloned()
```

Comment:
Nit: I believe this could be `copied` instead of `cloned`.

```suggestion
            .copied()
```

Reply:
Fixed 2 instances in 827069a22.